app/main.py

```diff
@@ -1,67 +1,212 @@
+import hashlib
 import json
+from datetime import datetime, timedelta, timezone
+from functools import lru_cache
+from pathlib import Path
 from typing import MutableMapping
 from uuid import UUID
 
 import uvicorn
 import yaml
 from fastapi import FastAPI, Query, HTTPException
+from pydantic import BaseModel
 from sdc.crypto.jwe_helper import JWEHelper
 from sdc.crypto.key_store import KeyStore
 
-KEY_PURPOSE_SDS = "supplementary_data"
-
 app = FastAPI()
 
 with open("dev-keys.yml", encoding="UTF-8") as keys_file:
     keys = KeyStore(yaml.safe_load(keys_file))
 
+KEY_PURPOSE_SDS = "supplementary_data"
+MOCK_DATA_ROOT_PATH = Path(__file__).parent.parent / "mock_data"
+
+# period_id to match with Launcher
+PERIOD_ID = "201605"
+
+# Currently only have a single schema version
+SCHEMA_VERSION = "v1.0.0"
+
+# Currently only have 1 reporting unit example for each schema version
+TOTAL_REPORTING_UNITS = 1
+
+FORM_TYPES = ["001"]
+
+# Hardcoded paths for now - update if any changes are made to https://github.com/ONSdigital/sds-schema-definitions/tree/main/examples
+MOCK_DATA_PATHS_BY_SURVEY_ID = {
+    "test": ["123"],
+    "prodcom": ["014"],
+    "bres_and_brs": [
+        "221",  # BRES
+        "241",  # BRS
+    ],
+    "roofing_tiles_slate": [
+        "068",  # Roofing tiles
+        "071",  # Slate
+    ],
+    "sand_and_gravel": [
+        "066",  # Sand and & Gravel (Land Won)
+        "076",  # Sand & Gravel (Marine Dredged)
+    ],
+}
+
+
+class DatasetMetadata(BaseModel):
+    """
+    Model for SDS Metadata Response
+    copied from https://github.com/ONSdigital/sds/blob/main/src/app/models/dataset_models.py#L19
+    """
+
+    # Required fields
+    survey_id: str
+    period_id: str
+    form_types: list[str]
+    sds_published_at: str
+    total_reporting_units: int
+    schema_version: str
+    sds_dataset_version: int
+    filename: str
+    dataset_id: UUID
+
+    # Optional fields
+    title: str | None = None
+
+
+class UnitData(BaseModel):
+    """
+    Model for SDS Unit Data Response
+    copied from https://github.com/ONSdigital/sds/blob/main/src/app/models/dataset_models.py#L53
+    """
+
+    dataset_id: UUID
+    survey_id: str
+    period_id: str
+    form_types: list[str]
+    schema_version: str
+    data: str
+
 
 @app.get("/v1/unit_data")
-def get_sds_data(
-    dataset_id: UUID, identifier: str = Query(min_length=1)
-) -> MutableMapping:
+def get_unit_data(dataset_id: UUID, identifier: str = Query(min_length=1)) -> UnitData:
     # The mock current does not make use of identifier
-    guid_filename_map = {
-        "c067f6de-6d64-42b1-8b02-431a3486c178": "supplementary_data",
-        "693dc252-2e90-4412-bd9c-c4d953e36fcd": "supplementary_data_v2",
-        "9b418603-ba90-4c93-851a-f9cecfbda06f": "supplementary_data_v3",
-    }
+    """Return an encrypted map of mocked unit data for the given dataset_id"""
+    _, dataset_to_unit_data_map = load_mock_data()
```

Comment:
I might have misunderstood the questions sorry, but with the LRU caching decorator on `load_mock_data()`, when either `/v1/unit_data` or `/v1/dataset_metadata` end point is first called, the `load_mock_data()` will cache the result for both of the mappings. Subsequent calls to `load_mock_data()` will reuse this cached result for both mappings. So the mappings are only generated once for the lifecycle of the mock SDS instance. 

I've checked the object ID references for both maps each time `load_mock_data()` is called, and they remain the same, so it might be okay if we then filter the result afterwards? This is the same pattern used in the mock CIR as well (although that was for a much larger scope of data as it was loading in all of the schemas)